spec/store_spec.rb

```diff
@@ -0,0 +1,75 @@
+RSpec.describe Store do
+  let (:aws_s3_client) { Aws::S3::Client.new(stub_responses: true) }
+  let (:s3_store) { Store.new }
+  
+  before do
+    allow(s3_store).to receive(:aws_client).and_return(aws_s3_client)
+  end
+  
+  context '#list_buckets' do
+    
+    it 'should list all the buckets in the linked account when response is not empty/nil' do
+      aws_s3_client.stub_responses(:list_buckets, buckets: [{ name: 'bucket1' }, { name: 'bucket2' }, { name: 'bucket3' }])
+      response = aws_s3_client.list_buckets
+      expect(response.buckets.count).to eq(3)
+      expect(STDOUT).to receive(:puts).thrice
+      s3_store.list_buckets
+    end
+    
+    it 'raise S3StoreEmptyError when there are no buckets - empty / nil response' do
+      aws_s3_client.stub_responses(:list_buckets, buckets: [])
+      expect { s3_store.list_buckets }.to raise_error(S3StoreEmptyError)
+    end
+  end
+
+  context '#create_bucket' do
+    
+    let (:bucket_name) { "foobar007" }
+    let (:invalid_bucket_name) { "TestBucket!" }
+    let (:response) { double('response') }
+    
+    before do
+      aws_s3_client.stub_responses(:create_bucket)
+    end
+
+    it 'should create a new bucket with bucket name' do
+      expect(aws_s3_client).to receive(:create_bucket).with({bucket: bucket_name}).and_return(response)
+      expect(STDOUT).to receive(:puts).once
+      s3_store.create_bucket(bucket_name)
+    end
+
+    it 'should raise Aws::S3::Errors::InvalidBucketName' do
+      expect { s3_store.create_bucket(invalid_bucket_name) }.to raise_error(S3StoreArgumentError)
+    end
+
+    it 'should rescue Aws::S3::Errors::BucketAlreadyOwnedByYou' do
+      allow(s3_store).to receive(:create_bucket).and_raise('Aws::S3::Errors::BucketAlreadyOwnedByYou')
```

Comment:
You an objects of `Aws::S3::Errors::BucketAlreadyOwnedByYou` class with the appropriate initializer values. Ref: https://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Errors/BucketAlreadyOwnedByYou.html

Reply:
The same can be applied everywhere instead of stub_responses..

Reply:
the link you provided is of version 2 which is outdated. Version 3 has the issue reported as mentioned in my comments above.